cmd/scaffold.go

```diff
@@ -1426,13 +1427,20 @@ func (fnb *FlowNodeBuilder) setRootSnapshot(rootSnapshot protocol.Snapshot) erro
 		return fmt.Errorf("failed to read root sealed result: %w", err)
 	}
 
-	sealingSegment, err := fnb.RootSnapshot.SealingSegment()
+	rootHeader, err := fnb.RootSnapshot.Head()
+	if err != nil {
+		return fmt.Errorf("could not read root header: %w", err)
+	}
+	rootBlockID := rootHeader.ID()
+	fnb.FinalizedRootBlock, err = fnb.Storage.Blocks.ByID(rootBlockID)
```

Comment:
oh, good idea!  We only depend on the finalized and sealed root block, so doesn't have to build the sealing segment again, which avoids the problematic validation. Nice 👍